### ISSUE TITLE
App crashing on launch after migrating Firebase from Cocoapods to Swift Package Manager

### ISSUE THREAD

**Issue by brandtdaniels** [firebase/firebase-ios-sdk#9398]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.2.1
  * Firebase SDK version: 8.12.1
  * Installation method: `Swift Package Manager` 
  * Firebase Component: Analytics, Crashlytics, Messaging, Performance, InAppMessaging-Beta, RemoteConfig
  * Target platform(s): `iOS`

#### Steps to reproduce:

Migrated Firebase from Cocoapods to Swift Package Manager, App crashes on launch.

#### Debug Log:

```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[FBLPromise HTTPBody]: unrecognized selector sent to instance 0x280b01d40'
terminating with uncaught exception of type NSException
```

![Screen Shot 2022-02-28 at 9 53 25 AM](https://user-images.githubusercontent.com/1013676/156033455-5df2b83e-adf3-420b-8e73-35025ce6d375.png)

**Comment by paulb777** [comment]
The `unrecognized selector` usually indicates a problem with project configuration.  Are you mixing CocoaPods and Swift PM?  See #9251 as a possible duplicate.

**Comment by brandtdaniels** [comment]
> The `unrecognized selector` usually indicates a problem with project configuration. Are you mixing CocoaPods and Swift PM? See #9251 as a possible duplicate.

Hi Paul, Yes we are mixing as we are migrating away from Cocoapods. I have determined that the issue is due to another Cocoapod we rely on that relies on MLKit Barcode Scanning libraries. It seems that Firebase and MLKit both rely on GoogleUtilities. Due to this symbiotic relationship, I have determined that I can not migrate Firebase to SPM until I migrate my dependency on MLKit to SPM, which I don't think MLKit yet supports SPM.
